src/Component/Tests/Symfony/Validator/State/ValidateProviderTest.php

```diff
@@ -0,0 +1,320 @@
+    {
+        $request = $this->prophesize(Request::class);
+        $attributes = $this->prophesize(ParameterBag::class);
+
+        $operation = new Create();
+
+        $context = new Context(new RequestOption($request->reveal()));
+
+        $this->decorated->provide($operation, $context)->willReturn(['foo' => 'fighters'])->shouldBeCalled();
+
+        $request->isMethodSafe()->willReturn(false);
+        $request->getRequestFormat()->willReturn('html');
+
+        $request->attributes = $attributes;
+
+        $attributes->get('form')->willReturn(null);
+
+        $attributes->set('is_valid', Argument::any())->shouldNotBeCalled();
+
+        $this->validateProvider->provide($operation, $context);
+    }
+
+    /** @test */
+    public function it_validates_resource_on_non_html_format(): void
+    {
+        $request = $this->prophesize(Request::class);
+        $attributes = $this->prophesize(ParameterBag::class);
+        $data = $this->prophesize(\stdClass::class);
+        $constraintViolationList = $this->prophesize(ConstraintViolationListInterface::class);
+
+        $request->getRequestFormat()->willReturn('json')->shouldBeCalled();
+
+        $operation = new Create();
+
+        $context = new Context(new RequestOption($request->reveal()));
+
+        $this->decorated->provide($operation, $context)->willReturn($data)->shouldBeCalled();
+
+        $request->attributes = $attributes;
+
+        $attributes->get('form')->willReturn(null);
+
+        $this->validator->validate($data, null, null)->willReturn($constraintViolationList)->shouldBeCalled();
+
+        $constraintViolationList->count()->willReturn(0)->shouldBeCalled();
+
+        $this->validateProvider->provide($operation, $context);
+    }
+
+    /** @test */
+    public function it_validates_resource_with_validation_context_on_non_html_format(): void
+    {
+        $request = $this->prophesize(Request::class);
+        $attributes = $this->prophesize(ParameterBag::class);
+        $data = $this->prophesize(\stdClass::class);
+        $constraintViolationList = $this->prophesize(ConstraintViolationListInterface::class);
+
+        $request->getRequestFormat()->willReturn('json')->shouldBeCalled();
+
+        $operation = new Create(validationContext: ['groups' => ['sylius']]);
+
+        $context = new Context(new RequestOption($request->reveal()));
+
+        $this->decorated->provide($operation, $context)->willReturn($data)->shouldBeCalled();
+
+        $request->attributes = $attributes;
+
+        $attributes->get('form')->willReturn(null);
+
+        $this->validator->validate($data, null, ['sylius'])->willReturn($constraintViolationList)->shouldBeCalled();
+
+        $constraintViolationList->count()->willReturn(0)->shouldBeCalled();
+
+        $this->validateProvider->provide($operation, $context);
+    }
+
+    /** @test */
+    public function it_throws_an_exception_when_validating_resource_on_non_html_format(): void
```

Comment:
> it_validates_resource_with_validation_context_on_non_html_format
> it_throws_an_exception_when_validating_resource_on_non_html_format

I wonder if methods names can or should be re-pharesed a bit.
My first thought is that difference in validation without exception and with exception is due "validation context" presence.

Though I don't have better idea